src/ios/BranchSDK.m

```diff
@@ -323,20 +311,17 @@ - (void)createBranchUniversalObject:(CDVInvokedUrlCommand*)command
 
   for (id key in properties) {
     if ([key isEqualToString:@"contentMetadata"]){
-      NSDictionary *metadata = (NSDictionary *)[properties valueForKey:key];
-
-      for (id key_ in metadata) {
-        [branchUniversalObj addMetadataKey:key_ value:[metadata valueForKey:key_]];
-      }
+        NSMutableDictionary<NSString *,NSString *> *metadata = (NSMutableDictionary<NSString *,NSString *> *)[properties valueForKey:key];
+        [[branchUniversalObj contentMetadata] setCustomMetadata:metadata];
     }
     else if ([key isEqualToString:@"contentIndexingMode"]) {
       NSString *indexingMode = [properties valueForKey:key];
       // Default contentIndexMode is always public
       if ([indexingMode isEqualToString:@"private"]) {
-        branchUniversalObj.contentIndexMode = BranchContentIndexModePrivate;
```

Comment:
Double check that this is actually a one to one replacement behavior. Cause if it's not we might need to change the key check as well.

Reply:
Yeah, based on my test setting to public/private properly adjusts the `publicly_indexable` field in requests.